pkg/config/network.go

```diff
@@ -58,3 +61,28 @@ func getClusterDNS(serviceCIDR string) (string, error) {
 
 	return dnsClusterIP.String(), nil
 }
+
+func (c *Config) EnsureNetworksDontOverlap() error {
+	linkList, err := netlink.LinkList()
```

Comment:
It should be possible to test the logic of this function without spinning up different VMs with "bad" network settings. If the public interface took no arguments, as this function does, and called a separate private function that took as arguments a function to return the list of links and a function to return the routes for a given link, then a unit test could provide fake implementations of those functions.